msal/src/main/java/com/microsoft/identity/client/NativeAuthPublicClientApplicationConfiguration.kt

```diff
@@ -0,0 +1,204 @@
+import java.io.Serializable
+
+/**
+ * Extends [PublicClientApplicationConfiguration] to add a few extra fields specific to Native Auth
+ * app implementation.
+ */
+@Accessors(prefix = ["m"])
+@Getter
+class NativeAuthPublicClientApplicationConfiguration :
+    PublicClientApplicationConfiguration(),
+    Serializable {
+    companion object {
+        private val TAG = NativeAuthPublicClientApplicationConfiguration::class.java.simpleName
+        private val VALID_CHALLENGE_TYPES = listOf("password", "oob", "redirect")
+    }
+
+    private object NativeAuthSerializedNames {
+        const val CHALLENGE_TYPES = "challenge_types"
+        const val USE_REAL_AUTHORITY = "useRealAuthority"
+        const val DC = "dc"
+    }
+
+    @SerializedName(NativeAuthSerializedNames.CHALLENGE_TYPES)
+    private var challengeTypes: List<String>? = null
+
+    @SerializedName(NativeAuthSerializedNames.USE_REAL_AUTHORITY)
+    var useRealAuthority: Boolean? = null
+
+    @SerializedName(NativeAuthSerializedNames.DC)
+    var dc: String? = null
+
+    fun getChallengeTypes(): List<String>? {
+        return challengeTypes
+    }
+
+    fun setChallengeTypes(challengeTypes: List<String>?) {
+        this.challengeTypes = challengeTypes
+    }
+
+    fun mergeConfiguration(config: NativeAuthPublicClientApplicationConfiguration) {
+        // Call super.mergeConfiguration to handle base configuration fields
+        super.mergeConfiguration(config)
+
+        // Native-auth specific account mode check. If dev passes MULTIPLE, we still want to set it so
+        // we can throw an exception to clarify that MULTIPLE account mode can't be used with Native
+        // Auth.
+        accountMode = if (config.accountMode != null) config.accountMode else accountMode
+
+        // Handle Native Auth specific fields
+        challengeTypes = if (config.challengeTypes == null) challengeTypes else config.challengeTypes
+    }
+
+    /**
+     * Override base validateConfiguration() method as native auth validation is different
+     * Had to make this function private as companion object in
+     * [NativeAuthPublicClientApplicationConfigurationFactory] cannot access it otherwise
+     */
+    public override fun validateConfiguration() {
+        // Check that a client id was passed
+        if (TextUtils.isEmpty(clientId)) {
+            throw MsalClientException(
+                MsalClientException.NATIVE_AUTH_USE_WITHOUT_CLIENT_ID_ERROR_CODE,
+                MsalClientException.NATIVE_AUTH_USE_WITHOUT_CLIENT_ID_ERROR_MESSAGE
+            )
+        }
+
+        // If we are given a redirectUri, we should run the superclass validation method to make sure
+        // we can use web auth
+        if (redirectUri != null) {
+            super.validateConfiguration()
+        } else {
+            LogSession.log(TAG, Logger.LogLevel.WARN, "No redirect URI was passed. You cannot use browser-based auth flow to handle BrowserRequired error.")
```

Comment:
I'm not sure if this log message needs to be updated, mainly because we're referencing BrowserRequired.